lesson_13/roman_alex/lesson_13_roman_alex.py

```diff
@@ -0,0 +1,59 @@
+from math import sqrt
+
+class PerimeterMixin:
+
+    def perimeter(self):
+        return f'Perimeter is {sum(self.sides)}'
+
+
+class Polygons(PerimeterMixin):
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return '{} is the number of sides'.format(no_of_sides)
+
+    def display(self):
+        for side_index, length in enumerate(self.sides, start=1):
+            print('Side {} with length: {}'.format(side_index, length))
+
+
+class Triangle(Polygons):
+    # `s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2`
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return (s_p * (s_p - s1) * (s_p - s2) * (s_p - s3)) ** 0.5
+
+
+class Square(Polygons):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return side ** 2
+
+    def from_area(self):
```

Comment:
The logic for getting the sides from the area is correct. We want this method to return a new Square object. Your method is only printing the sides and we can't use the object further.

Try to change the method in such a way that will return a Square object. For creating the new object we need the class object (cls) as an argument instead of self. What concept from Python we can use to achieve this?

Reply:
Code updated...